widgets/query-resources-by-category-widget.php

```diff
@@ -1,243 +1,269 @@
+    }
+
+    private function get_categories($post)
+    {
+        $categories_names = array();
+
+        if (isset($post)) :
+            global $wpdb;
+
+            $get_post_cat_ids = wp_get_post_categories($post->ID, array('fields' => 'ids'));
+            $table_name = $wpdb->prefix . "terms";
+
+            $cat_results = $wpdb->get_results(
+                $wpdb->prepare(
+                    "SELECT name
+                    FROM $table_name
+                    WHERE `term_id` IN (" . implode(', ', array_fill(0, count($get_post_cat_ids), '%d')) . ")
+                    ",
+                    $get_post_cat_ids
+                )
+            );
+
+            if ($cat_results) :
+                foreach ($cat_results as $key => $cat) {
+                    $categories_names[] = $cat->name;
+                }
+            endif;
+        endif;
+
+        if (isset($_GET['id'])) :
+            try {
+                $dataset = wpckan_api_package_show(wpckan_get_ckan_domain(), $_GET['id']);
+                $categories_names = $dataset['taxonomy'];
+            } catch (Exception $e) { #
+                wpckan_log($e->getMessage());
+            }
+        endif;
+
+        return $categories_names;
+    }
+
+    /**
+     * Outputs the content of the widget
+     *
+     * @param array $args
+     * @param array $instance
+     */
+    public function widget($args, $instance)
+    {
+        global $post;
+
+        $search_field               = isset($instance['search_field']) ? sanitize_text_field($instance['search_field']) : 'taxonomy';
+        $limit                      = isset($instance['limit']) ? sanitize_text_field($instance['limit']) : -1;
+        $categories_names           = $this->get_categories($post);
+        $template                   = isset($instance['template']) ? $instance['template'] : 'dataset-list';
+        $output_fields              = isset($instance['output_fields']) ? sanitize_text_field($instance['output_fields']) : 'title';
+        $output_fields_resources    = isset($instance['output_fields_resources']) ? sanitize_text_field($instance['output_fields_resources']) : '';
+
+        $filter_value = "(\"" . implode("\" OR \"", $categories_names) . "\")";
+
+        if (!empty($categories_names) && !(empty($search_field))) :
+            $shortcode = '[wpckan_query_datasets filter_fields=\'{"' . $search_field . '":"' . urlencode($filter_value) . '"}\'';
+            
+            if ($template !== "dataset-list") :
+                $shortcode .= ' template="' . $template . '"';
+            endif;
+
+            if (!empty($instance['organization']) && $instance['organization'] != '-1') :
+                $shortcode .= ' organization="' . $instance['organization'] . '"';
+            endif;
+
+            if (!empty($instance['type'])) :
+                $shortcode .= ' type="' . $instance['type'] . '"';
+            endif;
+
+            if (!empty($instance['limit']) && $instance['limit'] > 0) :
```

Comment:
Is that colon right?

Reply:
Are you referring to the colon on this line `if (!empty($instance['limit']) && $instance['limit'] > 0) :`?

If it is the line that you are referring to, that colon is right. It is the **if else statement colon style in PHP**

Reply:
So that was probably a parse error before. 

Is there a reason that you're not using the standard `if () {} else {}` that's basically in all the c-syntax like languages?

Reply:
One of the main reasons for using this syntax is the readability of the code as in WordPress coding we use both `HTML`  and `PHP` written together in the code.

As there are multiple contributors to this theme development, each of them has different preferences for syntax style.

This style of writing is mentioned in both PHP and WP documentation:
- https://www.php.net/manual/en/control-structures.alternative-syntax.php
- https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/